pkg/gce-pd-csi-driver/node.go

```diff
@@ -74,10 +73,10 @@ func (ns *GCENodeServer) NodePublishVolume(ctx context.Context, req *csi.NodePub
 		return nil, status.Error(codes.InvalidArgument, "NodePublishVolume Volume Capability must be provided")
 	}
 
-	if _, alreadyExists := ns.volumes.LoadOrStore(volumeID, true); alreadyExists {
-		return nil, status.Error(codes.Aborted, fmt.Sprintf("An operation with the given Volume ID %s already exists", volumeID))
+	if acquired := ns.volumeLocks.TryAcquire(volumeID); !acquired {
```

Comment:
do we allow to mount the same volume to different paths concurrently?

Reply:
No, in our final design we decided not to support that to be safe because the cost of a bind mount is very small.